Assets/02. Scripts/02-06. Structure/Furniture/Furniture.cs

```diff
@@ -1,12 +1,92 @@
+using Photon.Pun;
+using System;
 using UnityEngine;
 
-public class Furniture : MonoBehaviour
+public class Furniture : MonoBehaviour, IGridItemHandler
 {
-    private FurnitureData _data;
-    public FurnitureData Data => _data;
+    [SerializeField]
+    FurnitureStat _stat;
```

Comment:
접근 제한자 추가해주세요! 그리고 SerializeField도 필요없지 않나요?

Reply:
```suggestion
    private FurnitureStat _stat;
    public FurnitureStat Stat => _stat;
```
제네릭 인터페이스에 TStat 타입을 뺀다면, 이렇게 TClass 타입이 될 클래스 내부 스탯 필드에 접근할 수 있는 프로퍼티를 추가해주면 됩니다.

Reply:
접근 제한자 추가했고 SerializeField는 씬에서 변수 잘 바뀌는지 확인하려고 넣었습니다. 그리고 TStat부분 은닉화를 유지하기 위해 넣었습니다. 이전에도 매개변수에 Stat을 같이 넘겨준 이유가 Stat 내부 변수들이 인터페이스 구현 클래스에서 변경되어야 하기 때문에 전부 public이라, stat클래스는 private으로 하고 매개변수로만 넘겨주게 하였습니다.

Reply:
"Stat 클래스가 구현된 인터페이스 메서드 내부에서만 변경되도록 하기 위해서"로 이해했는데 맞나요? 은닉화를 유지하기 위해서라는건 잘 모르겠네요. 오히려 은닉화를 지향하신다면 제네릭 인터페이스에서 TStat을 빼는게 맞습니다. 인터페이스 선언부에서 TStat의 존재가 드러나지 않을테니까요. 만약 첫번째 이유라면 지금 설계 그대로 유지하시면 될 것 같고, 은닉화를 좀 더 지향하고 싶다면 제네릭 인터페이스에서 TStat을 빼는 쪽으로 가시는 것이 좋을 것 같습니다.

Reply:
아 은닉화에 대해 말씀하신게 서로 다른거 같습니다. 외부에서 내부 접근에 대한 은닉화를 말하는거고 인터페이스 구현 클래스 생성 자체는 어쩌피 Manager가 도맡아 하기 때문에 그부분은 생각하지 않았습니다. Machine을 예로 들면 Machine 클래스에 있는 Stat을 IInteractable 인터페이스를 구현한 TryInteract 함수로 Machine과 MachineStat을 보내주잖아요? 이 이유가 Machine 클래스 외부에서 Stat 클래스에 접근하지 못하도록 하기 위함입니다. Stat 내부 변수들이 public인 데다가 Machine의 이 함수들을 호출하는 곳은 Player 클래스라 Stat이 은닉화가 안되어있으면 Player에서 접근 및 수정이 가능합니다.

제네릭 인터페이스에서 TStat을 빼게 되더라도 stat을 public으로 바꾸지 말고 지금 Furniture, Machine, Storage가 모두 __Stat클래스를 가지고 있으니까 IStat으로 빼서 매개변수에 넣어야 된다고 생각합니다.

Reply:
이해했습니다. 이대로 가시죠